banners/english/components/BannerCtrl.vue

```diff
@@ -116,7 +117,11 @@ watch( contentState, async () => {
 } );
 
 function onCloseMain(): void {
-	contentState.value = ContentStates.SoftClosing;
+	if ( props.remainingImpressions > 0 ) {
```

Comment:
Seeing this repeated across banners makes my DRY reflexes twitch. Mostly because I'd like to give this behavior a name like `softCloseIfNotLastImpression`, etc. OTOH it would make the code less readable and clear, because this if statement is already quite clear and we'd have to pass `remainingImpressions`, `contentState` *and* `onClose` into that "generalized" function to keep it generalized.

Unless we find a way to share `contentState` (and maybe some change/watch methods) across channels (maybe we only have two versions of this, one for desktop and one for mobile), I'd say leave those `if` statements in. And maybe even leave an eventual extraction of `contentState` to another actual banner ticket.

Reply:
Yeah, my thought process went through the same sequences. I ended up putting it into each banner because even though the functionality is shared it's still something that happens per-banner.